x-pack/test/accessibility/apps/automated.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Fs from 'fs';
+import Yaml from 'js-yaml';
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../ftr_provider_context';
+
+const PATH = '../x-pack/test/accessibility/a11y_errors_count.yml';
+
+interface Errors {
+  [key: string]: number;
+}
+
+function readErrors() {
+  let yaml;
+  try {
+    yaml = Fs.readFileSync(PATH, 'utf8');
+  } catch (error) {
+    if (error.code !== 'ENOENT') {
+      throw error;
+    }
+  }
+
+  return yaml ? Yaml.safeLoad(yaml) : {};
+}
+
+function updateErrors(originalErrors: Errors, currentErrors: Errors) {
+  const updatedErrors: Errors = {};
+
+  for (const [key, value] of Object.entries(currentErrors)) {
+    const originalCount = originalErrors[key];
+
+    if (originalCount && originalCount < value) {
+      updatedErrors[key] = originalCount;
+    } else {
+      updatedErrors[key] = value;
+    }
+  }
+
+  Fs.writeFileSync(PATH, Yaml.safeDump({ errorsPerPage: updatedErrors }));
+}
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const { common, header } = getPageObjects(['common', 'header']);
+  const a11y = getService('a11y');
+  const config = getService('config');
+  const log = getService('log');
+
+  let originalErrors = {};
+  const currentErrors: Errors = {};
+
+  describe('Automated pass', () => {
+    before(async () => {
+      await common.navigateToActualUrl('home');
+      originalErrors = readErrors();
+    });
+    after(() => {
+      updateErrors(originalErrors, currentErrors);
+    });
+
+    it('runs', async () => {
+      const apps: Array<[string, { pathname: string; hash?: string }]> = Array.from(
+        Object.entries(config.get('apps'))
+      );
+
+      for (const [app, { hash }] of apps) {
+        if (app === 'login') continue; // causes a perpetual login loop, ignore
+        await common.navigateToActualUrl(app, hash);
+        await header.waitUntilLoadingHasFinished();
+        const errorCount = await a11y.testAppSnapshot({
```

Comment:
I feel like `errorCount` is a pretty loose error to track, maybe we can utilize snapshots here to store a snapshot of the error report (assuming it's stable). Since this is in a loop we would need to use named snapshots, but that should be supported right @dgieselaar?

Reply:
I haven't tested it, but I think I'm just passing the arguments directly to Jest, so it might just work. If not I would assume it's fairly easy to implement (🤞).

Reply:
Sounds pretty reasonable - will give it a shot 👍

Reply:
Looking deeper into this and I'm not sure if the results are stable enough to reasonably do this. 

I should say, the results from `a11y.testAppSnapshot()` are consistent but there are a lot of ways for FUD to creep in. For example, changing the order of two elements on a page (one of which that is failing) changes the snapshot because the dom order changed. Changing just about anything in an element that's failing also causes a change in the snapshot... 

I mean, it'd be great if folks went and fixed the issues then but I worry about people being thrown off by seemingly unrelated test failures.

By giving just a count, we might catch a little less but there's high confidence that the dev won't be taken by surprise that they have a failing test.

Reply:
Interesting, do you have a copy of a report that I could look at? Wondering if we could reduce it to just the relevant info and sort it or something...